independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/StringUtil.java

```diff
@@ -0,0 +1,89 @@
+                if (Character.isUpperCase(c)) {
+                    // an uppercase-starting word
+                    idx = str.offsetByCodePoints(idx, 1);
+                    if (idx < str.length()) {
+                        c = str.codePointAt(idx);
+                        if (Character.isUpperCase(c)) {
+                            // all-caps word; need one look-ahead
+                            int nextIdx = str.offsetByCodePoints(idx, 1);
+                            while (nextIdx < str.length()) {
+                                c = str.codePointAt(nextIdx);
+                                if (Character.isLowerCase(c)) {
+                                    // ended at idx
+                                    return str.substring(start, idx);
+                                }
+                                idx = nextIdx;
+                                nextIdx = str.offsetByCodePoints(idx, 1);
+                            }
+                            // consumed the whole remainder, update idx to length
+                            idx = str.length();
+                            return str.substring(start);
+                        } else {
+                            // initial caps, trailing lowercase
+                            idx = str.offsetByCodePoints(idx, 1);
+                            while (idx < str.length()) {
+                                c = str.codePointAt(idx);
+                                if (Character.isUpperCase(c)) {
+                                    // end
+                                    return str.substring(start, idx);
+                                }
+                                idx = str.offsetByCodePoints(idx, 1);
+                            }
+                            // consumed the whole remainder
+                            return str.substring(start);
+                        }
+                    } else {
+                        // one-letter word
+                        return str.substring(start);
+                    }
+                } else {
+                    // a lowercase-starting word
+                    idx = str.offsetByCodePoints(idx, 1);
+                    while (idx < str.length()) {
+                        c = str.codePointAt(idx);
+                        if (Character.isUpperCase(c)) {
+                            // end
+                            return str.substring(start, idx);
+                        }
+                        idx = str.offsetByCodePoints(idx, 1);
+                    }
+                    // consumed the whole remainder
+                    return str.substring(start);
+                }
+            }
+        };
+    }
+
+    public static <T> Iterable<T> iteratorToIterable(Iterator<T> iterator) {
+        return () -> iterator;
+    }
+
+    public static String hyphenate(final String orig) {
+        return String.join("-", iteratorToIterable(camelHumpsIterator(orig)));
```

Comment:
```suggestion
        return String.join("-", () -> camelHumpsIterator(orig));
```

Reply:
I think we need the return type to be `Iterable<String>`

```   
public static String hyphenate(final String orig) {
  return String.join("-", (Iterable<String>) () -> camelHumpsIterator(orig));
}
```